org.civicrm.reqangular/test/components/heatmap.component.spec.js

```diff
@@ -0,0 +1,108 @@
+/* eslint-env amd, jasmine */
+
+define([
+  'common/angular',
+  'common/lodash',
+  'common/components/heatmap.component'
+], function (angular, _) {
+  describe('Heatmap component', function () {
+    var $componentController, cmp;
+
+    beforeEach(function () {
+      module('common.components');
+      inject(function (_$componentController_) {
+        $componentController = _$componentController_;
+      });
+
+      cmp = $componentController('heatmap');
+    });
+
+    it('must be defined', function () {
+      expect(cmp).toBeDefined();
+    });
+
+    describe('on init', function () {
+      it('sets days equal to an empty array', function () {
+        expect(cmp.days).toEqual([]);
+      });
+    });
+
+    describe('converting values to days and heat range', function () {
+      var expectedDays;
+
+      beforeEach(function () {
+        var values = {
+          0: 0,
+          1: 2,
+          2: 2,
+          3: 10,
+          4: 6,
+          5: false,
+          6: false
+        };
+
+        expectedDays = getExpectedDays([
+          { value: 0, heat: 'low' },
+          { value: 2, heat: 'low' },
+          { value: 2, heat: 'low' },
+          { value: 10, heat: 'high' },
+          { value: 6, heat: 'medium' },
+          { value: 0, heat: 'disabled' },
+          { value: 0, heat: 'disabled' }
+        ]);
+
+        cmp.values = values;
+        cmp.$onChanges({ values: { currentValue: values } });
```

Comment:
`$onChanges` is triggered automatically when the app is running, but not when running tests. That's the reason for the helper function `controllerOnChanges` in L&A. 

Since this was a single component, I didn't make a generic `controllerOnChanges` that could be used in any extension. If the need to stub these calls we might consider using [angular-stub-changes](https://github.com/abyx/angular-stub-changes).

More info:
http://www.codelord.net/2017/01/17/testing-components-with-%24onchanges-using-angular-stub-changes/